.github/workflows/build.yml

```diff
@@ -34,7 +34,7 @@ jobs:
       run: pip install poetry
 
     - name: Build python-build
-      run: python -m build -x
+      run: python -m build -x -n
```

Comment:
Not really related to this change, but what is the purpose of this test? Who would want to build this package by manually using `pip` to install its build-time dependencies, then building with isolation disabled?

The kind of people who would want build isolation would presumably want to use the no-dependencies / minimal dependencies bootstrap mode, rather than installing `poetry` and `typing` and such, no?

Also, this file in general seems a bit fragile. It's hard-coding the build-time dependencies of a bunch of projects rather than pulling those dependencies from the respective projects' `build-system.requires` file.

Anyway, just hopefully some constructive criticism, not anything that needs to block this particular set of changes.

Reply:
With `-n`, it does not perform any isolation and does not use `pip` to install the dependencies.

This test just makes sure that the `python -m build` actually works. We mock too much in `pytest` and I feel much more comfortable having this extra tests. I Selected 4 packages (`python-build`, `dateutil`, `pip` and `Solaar`), to verify the command actually works, this particular patch changes two of them to the no isolation mode. Building `python-build` via poetry and without isolation does not have any particular meaning.

And yes, it would be safer to fetch `build-system.requires`. It should come in a separate PR. Just a thought but would it make sense to introduce a command to read the dependencies?

Reply:
> With `-n`, it does not perform any isolation and does not use `pip` to install the dependencies.

Yeah, I meant that you manually use `pip` to install the dependencies. For `python-dateutil`, `Solaar` and `pip` I could see that being a reasonable thing to do, but the only people who would ever want to do this would be distro packagers, and for `python-build` specifically, they're unlikely to want to because it's intended to be one of the first things you bootstrap.